service/storage_proxy.hh

```diff
@@ -433,7 +433,7 @@ public:
             scheduling_group_key stats_key, gms::feature_service& feat, const locator::shared_token_metadata& stm, locator::effective_replication_map_factory& erm_factory, netw::messaging_service& ms);
     ~storage_proxy();
 
-    remote& remote();
+    remote& remote() const;
```

Comment:
Why is member returning a mutable reference to a member const?

Reply:
My laziness. I thought why the hell not, it's an internal thing anyway.

Not sure what to do - I want to be able to do `remote().gossiper()` in `const` members of `storage_proxy` (note that `gossiper()` is `const` in `remote`). I don't want to duplicate the contents of `remote()` by having two versions (`const` and non-`const`)... well, most of the content is comments, I could leave them in one version and not the other, but meh.

Reply:
This is how `remote()` looks like on master:
```
2619 struct storage_proxy::remote& storage_proxy::remote() {                                                                                                                                                                                                  
   1     return *_remote;                                                                                                                                                                                                                                     
   2 } 
```

Doesn't look too hard to duplicate. I won't insist, but I do think it would be ever so slightly more correct. I agree it is not a big thing, and the question of whether a `const` member can return mutable pointers/references into the object is a highly flammable subject, that doesn't have a consensus in the C++ community AFAIK.